docs/integration.rst

```diff
@@ -866,3 +866,206 @@ These integrations allow you to perform various operations within various servic
+     - :doc:`How to use <howto/operator/bash>`
+     -
+     - :mod:`airflow.operators.bash_operator`
+     - :mod:`airflow.contrib.sensors.bash_sensor`
+
+   * - `Kubernetes <https://kubernetes.io/>`__
+     - :doc:`How to use <howto/operator/kubernetes>`
+     -
+     - :mod:`airflow.contrib.operators.kubernetes_pod_operator`
+     -
+
+   * - `Microsoft SQL Server (MSSQL) <https://www.microsoft.com/pl-pl/sql-server/sql-server-downloads>`__
+     -
+     - :mod:`airflow.hooks.mssql_hook`
+     - :mod:`airflow.operators.mssql_operator`
+     -
+
+   * - `MondoDB <https://www.mongodb.com/what-is-mongodb>`__
+     -
+     - :mod:`airflow.contrib.hooks.mongo_hook`
+     -
+     - :mod:`airflow.contrib.sensors.mongo_sensor`
+
+
+   * - `MySQL <https://www.mysql.com/products/>`__
+     -
+     - :mod:`airflow.hooks.mysql_hook`
+     - :mod:`airflow.operators.mysql_operator`
+     -
+
+   * - `OpenFaaS <https://www.openfaas.com/>`__
+     -
+     - :mod:`airflow.contrib.hooks.openfaas_hook`
+     -
+     -
+
+   * - `Oracle <https://www.oracle.com/pl/database/>`__
+     -
+     - :mod:`airflow.hooks.oracle_hook`
+     - :mod:`airflow.operators.oracle_operator`
+     -
+
+   * - `Papermill <https://github.com/nteract/papermill>`__
+     - :doc:`How to use <howto/operator/papermill>`
+     -
+     - :mod:`airflow.operators.papermill_operator`
+     -
+
+   * - `PostgresSQL <https://www.postgresql.org/>`__
+     -
+     - :mod:`airflow.hooks.postgres_hook`
+     - :mod:`airflow.operators.postgres_operator`
+     -
+
+   * - `Presto <http://prestodb.github.io/>`__
+     -
+     - :mod:`airflow.hooks.presto_hook`
+     - :mod:`airflow.operators.presto_check_operator`
+     -
+
+   * - `Python <https://www.python.org>`__
+     -
+     -
+     - :mod:`airflow.operators.python_operator`
+     - :mod:`airflow.contrib.sensors.python_sensor`
+
+   * - `Redis <https://redis.io/>`__
+     -
+     - :mod:`airflow.contrib.hooks.redis_hook`
+     - :mod:`airflow.contrib.operators.redis_publish_operator`
+     - :mod:`airflow.contrib.sensors.redis_pub_sub_sensor`
+       :mod:`airflow.contrib.sensors.redis_key_sensor`
+
+   * - `Samba <https://www.samba.org/>`__
+     -
+     - :mod:`airflow.hooks.samba_hook`
+     -
+     -
+
+   * - `SQLite <https://www.sqlite.org/index.html>`__
+     -
+     - :mod:`airflow.hooks.sqlite_hook`
+     - :mod:`airflow.operators.sqlite_operator`
+     -
+
+.. _protocol:
+
+Protocol integrations
+---------------------
+
+Operators and Hooks
+'''''''''''''''''''
+
+Protocol operators and hooks
+""""""""""""""""""""""""""""
+
+These integrations allow you to perform various operations within various services using standardized
+communication protocols or interface.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Service name
+     - Guide
+     - Hook
+     - Operators
+     - Sensors
+
+   * - `Internet Message Access Protocol (IMAP) <https://tools.ietf.org/html/rfc3501>`__
+     -
+     - :mod:`airflow.contrib.hooks.imap_hook`
+     -
```

Comment:
```suggestion
     - :mod: `airflow.contrib.operators.imap_attachment_to_s3_operator`
```

Reply:
It's a transport operator. I would like to put transfer operators in separate table. Generally, each group, e.g. Apache, Azure, AWS, Service, Software, Protocol is divided into subgroups - operators and hooks, Key management Logging, etc. Operators and hooks is divided in two groups - service/software/etc operators and hooks and transfer operators.  There is still no overview in each explanatory section, because for now I want to focus on the tables, and in the next step complete additional descriptions.  I try to make changes to this file in small steps.

Reply:
I am currently checking if operators are missing using automatic scripts.
```
grep -R -i 'This module is deprecated.' airflow | cut -d ":" -f 1 | sort |uniq | cut -d "." -f 1 | sed "s#/#.#g" > dep.txt
find airflow/{,gcp/,contrib/}{operators,sensors,hooks} | sort |grep -v "__init__" | grep ".py"| cut -d "." -f 1 | sed "s#/#.#g" |sort | uniq |grep -v "__pycache__"| grep -vf dep.txt > a.txt
cat docs/integration.rst  | grep ":mod:" | cut -d '`' -f 2 | sort | uniq |grep -v "__pycache__"  > b.txt
diff a.txt b.txt
```
For now, script results must be supervised. In the last refactoring step I will try to prepare a script that will automatically check if operators are missing.

Reply:
> It's a transport operator. I would like to put transfer operators in separate table.

Forgot that. Agree.

> grep -R -i 'This module is deprecated.' airflow | cut -d ":" -f 1 | sort |uniq | cut -d "." -f 1 | sed "s#/#.#g" > dep.txt
> find airflow/{,gcp/,contrib/}{operators,sensors,hooks} | sort |grep -v "__init__" | grep ".py"| cut -d "." -f 1 | sed "s#/#.#g" |sort | uniq |grep -v "__pycache__"| grep -vf dep.txt > a.txt
> cat docs/integration.rst  | grep ":mod:" | cut -d '`' -f 2 | sort | uniq |grep -v "__pycache__"  > b.txt
> diff a.txt b.txt

What a script :D 👍 

> In the last refactoring step I will try to prepare a script that will automatically check if operators are missing.

Sounds great! :)

Reply:
Fantastic!